Fixes around creating new connections

Crack opened the pull request:
A few fixes separated into a few commits:
* pass error from socket when connector fails to select a database, so that "timeout" error isn't hidden away from the user (without this we can get a connection with closed socket and not know about that until we try to use it, it may happen when connection gets broken outside of nginx)
* slave connection: use `set_keepalive` only when we are sure that Sentinel responded correctly - it may have timed out, and we don't want to put such connection in pool
* master connection: use `set_keepalive` always when Sentinel responded correctly, regardless of result of connecting to indicated Redis instance

pintsized commented:
Hey, thanks for this. Seems these changes break the tests though, could you try running them on your side? You'll need to `make start_redis_instances`, followed by `make test`.

Here's what I'm getting: https://gist.github.com/pintsized/bff6c6beaa4799762788a32fcd1a8884

Crack commented:
Oops... sorry for that, fixed.

There were two issues, both in my first commit:
1. I removed logging of an error message when AUTH command failed, and there was a test that looked for it.
2. My original fix to error returned from `:connect` didn't account for Sentinels: `:connect` tries to set database, which is an unsupported operation in Redis Sentinel, and possibly that's why error handling of that call was originally omitted.

Crack commented:
Hello, @pintsized, were you able to look into this MR?

pintsized commented:
Hey, I'm really sorry I seem to have totally forgotten about this. I managed to get travis CI working on this repo, if you rebase it should pick that up and build your pr.

Otherwise, yep LGTM. Looks like there's a 2-space indentation on line 363 if you feel like making it perfect ;)

Crack commented:
Invalid formatting is a perfectly valid issue, rebased and fixed :)
